sdk/eventhub/event-hubs/src/managementClient.ts

```diff
@@ -306,31 +309,95 @@ export class ManagementClient extends LinkEntity {
     options?: { retryOptions?: RetryOptions; abortSignal?: AbortSignalLike; requestName?: string }
   ): Promise<any> {
     try {
-      log.mgmt(
-        "[%s] Acquiring lock to get the management req res link.",
-        this._context.connectionId
-      );
-      await defaultLock.acquire(this.managementLock, () => {
-        return this._init();
-      });
-
       if (!options) {
         options = {};
       }
 
-      const sendRequestOptions: SendRequestOptions = {
-        maxRetries: options.retryOptions && options.retryOptions.maxRetries,
-        abortSignal: options.abortSignal,
-        requestName: options.requestName,
-        timeoutInSeconds: getRetryAttemptTimeoutInMs(options.retryOptions) / 1000,
-        delayInSeconds:
-          options.retryOptions &&
-          options.retryOptions.retryInterval &&
-          options.retryOptions.retryInterval >= 0
-            ? options.retryOptions.retryInterval / 1000
-            : undefined
+      const sendOperationPromise = () =>
+        new Promise<Message>(async (resolve, reject) => {
+          try {
+            const actionAfterTimeout = () => {
+              const address = this._mgmtReqResLink!.receiver.address || "address";
+              const desc: string =
+                `The request with message_id "${request.message_id}" to "${address}" ` +
+                `endpoint timed out. Please try again later.`;
+              const e: Error = {
+                name: "OperationTimeoutError",
+                message: desc
+              };
+              return reject(translate(e));
+            };
+
+            const waitTimer = setTimeout(
+              actionAfterTimeout,
+              getRetryAttemptTimeoutInMs(options!.retryOptions)
+            );
+
+            const initOperationStartTime = Date.now();
+
+            if (!this._isMgmtRequestResponseLinkOpen()) {
+              log.mgmt(
+                "[%s] Acquiring lock to get the management req res link.",
+                this._context.connectionId
+              );
+
+              await defaultLock.acquire(this.managementLock, () => {
+                return this._init().catch((err: Error) => {
+                  clearTimeout(waitTimer);
+                  reject(translate(err));
+                });
+              });
+            }
+            const initOperationEndTime = Date.now();
+
+            if (!options) {
+              options = {};
+            }
+
+            const operationTimeoutInMs =
+              options && options.retryOptions && options.retryOptions.timeoutInMs
+                ? options.retryOptions.timeoutInMs
+                : Constants.defaultOperationTimeoutInSeconds * 1000;
```

Comment:
Don't you have `getRetryAttemptTimeoutInMs()` to get the total operation timeout?